lib/Ix/DBIC/ResultSet.pm

```diff
@@ -395,7 +368,7 @@ sub ix_create ($self, $ctx, $to_create) {
     };
 
     if ($row) {
-      my @defaults = grep {; ! exists $user_prop->{$_} } keys %default_properties;
+      my @defaults = grep {; ! exists $this->{$_} } keys %default_properties;
```

Comment:
We can't check "$user_prop" (now $properties) anymore because it contains defaults as well, so we must look at the original input.